src/dataset-builder/ConceptSearch.test.ts

```diff
@@ -0,0 +1,152 @@
+const TestSearchInput = (props) => h(SearchInput, props);
+jest.mock('src/components/input', () => {
+  return {
+    ...jest.requireActual('src/components/input'),
+    DelayedSearchInput: TestSearchInput,
+  };
+});
+
+jest.mock('src/libs/ajax/GoogleStorage');
+type DataRepoExports = typeof import('src/libs/ajax/DataRepo');
+jest.mock('src/libs/ajax/DataRepo', (): DataRepoExports => {
+  return {
+    ...jest.requireActual('src/libs/ajax/DataRepo'),
+    DataRepo: jest.fn(),
+  };
+});
+describe('ConceptSearch', () => {
+  const onCancel = jest.fn();
+  const onCommit = jest.fn();
+  const onOpenHierarchy = jest.fn();
+  const actionText = 'action text';
+  const datasetId = '0';
+  const initialCart: SnapshotBuilderConcept[] = [];
+  const domainOption = dummyDatasetModel()!.snapshotBuilderSettings!.domainOptions[0];
+  const renderSearch = (initialSearch = '') => {
+    render(
+      h(ConceptSearch, {
+        actionText,
+        initialSearch,
+        initialCart,
+        onCancel,
+        onCommit,
+        onOpenHierarchy,
+        datasetId,
+        domainOption,
+      })
+    );
+  };
+
+  const displayedConcepts = [dummyGetConceptForId(102), dummyGetConceptForId(103)];
+  const mockSearch = jest.fn();
+
+  const mockDataRepoContract: DataRepoContract = {
+    dataset: (_datasetId) =>
+      ({
+        searchConcepts: mockSearch,
+      } as Partial<DataRepoContract['dataset']>),
+  } as Partial<DataRepoContract> as DataRepoContract;
+  asMockedFn(DataRepo).mockImplementation(() => mockDataRepoContract as DataRepoContract);
+  mockSearch.mockResolvedValue({
+    result: displayedConcepts,
+  });
+
+  it('renders the concept search with an initial search result', async () => {
+    // Arrange
+    renderSearch();
+    // Assert
+    expect(await screen.findByText(displayedConcepts[0].name)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[0].id)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[1].name)).toBeTruthy();
+    expect(await screen.findByText(displayedConcepts[1].id)).toBeTruthy();
+    // Action text not visible until a row is selected.
+    expect(screen.queryByText(actionText)).toBeFalsy();
+  });
+
+  it('searches based on user input', async () => {
+    // Arrange
+    renderSearch();
+    const searchText = 'search text';
+    // Act
+    const user = userEvent.setup();
+    await user.type(screen.getByPlaceholderText('Search'), searchText);
+    // Assert - the search is called with the search text.
+    expect(mockSearch).toHaveBeenCalledWith(domainOption.root, searchText);
+  });
+
+  it('filters based on search text', async () => {
+    // Arrange
+    renderSearch(displayedConcepts[0].name);
+    // Act
+    // Assert
+    expect(screen.findByText(displayedConcepts[0].name)).toBeTruthy();
+    // The second concept should not be displayed.
+    expect(() => screen.getByText(displayedConcepts[1].name)).toThrow();
```

Comment:
why does this throw instead of just toBeFalsy() like line 73?

Reply:
This test was incorrect; the page isn't filtering at all since the mocked search API isn't filtering. I'm going to replace it with a test that makes more sense.